test/unit/app/models/statistics_announcement_test.rb

```diff
@@ -292,6 +270,107 @@ class StatisticsAnnouncementTest < ActiveSupport::TestCase
+    announcement = create(:statistics_announcement, publication_type_id: PublicationType::NationalStatistics.id, publication: national_statistics)
+
+    announcement.delete
+    assert_equal Publication.find(national_statistics.id).title, "Test"
+    assert_equal StatisticsAnnouncement.where(id: announcement.id).count, 0
+  end
+
+  # === BEGIN: Publication Type update callback ===
+  test "should allow updating publication type with no attached publication" do
+    announcement = create(:statistics_announcement, publication_type_id: PublicationType::OfficialStatistics.id)
+
+    announcement.publication_type_id = PublicationType::NationalStatistics.id
+    announcement.save!
+
+    assert announcement.valid?
+    assert announcement.errors.empty?
+  end
+
+  test "should allow updating to same publication type with attached draft publication" do
+    national_statistics = create(:draft_national_statistics)
+    announcement = create(:statistics_announcement, publication_type_id: PublicationType::NationalStatistics.id, publication: national_statistics)
+
+    announcement.assign_attributes(
+      publication_type_id: PublicationType::NationalStatistics.id,
+      title: "New title",
+    )
+    announcement.save!
+
+    assert announcement.valid?
+    assert announcement.errors.empty?
+    assert_equal PublicationType::NationalStatistics.id, national_statistics.reload.publication_type_id
+  end
+
+  test "should allow updating to different publication type to attached publication and updates the draft publication" do
+    national_statistics = create(:draft_national_statistics)
+    announcement = create(:statistics_announcement, publication_type_id: PublicationType::NationalStatistics.id, publication: national_statistics)
+
+    announcement.assign_attributes(
+      publication_type_id: PublicationType::OfficialStatistics.id,
+      title: "New title",
+    )
+    announcement.save!
+
+    assert announcement.valid?
+    assert announcement.errors.empty?
+    assert_equal PublicationType::OfficialStatistics.id, national_statistics.reload.publication_type_id
+  end
+
+  test "should not allow creating new announcement with publication of mismatched Publication type" do
+    national_statistics = create(:draft_national_statistics)
+    announcement = build(:statistics_announcement, publication_type_id: PublicationType::NationalStatistics.id, publication: national_statistics)
+
+    announcement.assign_attributes(
+      publication_type_id: PublicationType::OfficialStatistics.id,
+      title: "New title",
+    )
+
+    assert_not announcement.save
+    assert announcement.errors[:publication].include? "type does not match announcement type: must be 'Official Statistics'"
+    assert_equal PublicationType::NationalStatistics.id, national_statistics.reload.publication_type_id
+  end
+
+  test "should not allow updating Publication type with attached published publication" do
```

Comment:
"[...] Publication type, when connected publication is published"?

Maybe we can use "connected" over "attached" in all these tests 🤔